# Movie Rating Prediction Model

## Khám phá dữ liệu phục vụ cho việc tiền xử lý

### Import các thư viện

In [2]:
import numpy as np
import pandas as pd

In [3]:
!pwd

/home/duyhoang/Downloads/TH NMKHDL/IMDB /notebooks


### Load vào dữ liệu thô sau khi đã được crawl

In [4]:
file_path = "../data/raw/IMDB_movies_final.csv"
df = pd.read_csv(file_path)

rank                                title  \
0          1                         Frankenstein   
1          2             One Battle After Another   
2          3                      The Running Man   
3          4                   Predator: Badlands   
4          5                             Playdate   
...      ...                                  ...   
10043  10046        Return of the Living Dead III   
10044  10047                            Sweetness   
10045  10048                         Dark Harvest   
10046  10049  Across the River and Into the Trees   
10047  10050                                Below   

                                    genres                    budget  \
0           Drama, Fantasy, Horror, Sci-Fi  $120,000,000 (estimated)   
1           Action, Crime, Drama, Thriller  $130,000,000 (estimated)   
2      Action, Adventure, Sci-Fi, Thriller  $110,000,000 (estimated)   
3      Action, Adventure, Sci-Fi, Thriller  $105,000,000 (estimated)   
4                           Action, Comedy                   $17,525   
...                                    ...                       ...   
10043                               Horror    $2,000,000 (estimated)   
10044                             Thriller                       NaN   
10045                      Fantasy, Horror                       NaN   
10046                  Drama, Romance, War                       NaN   
10047    Action, Horror, Mystery, Thriller   $40,000,000 (estimated)   

       release_year run_time    mpa  metascore vote_count  rating  
0            2025.0   2h 29m      R       77.0     (132K)     7.6  
1            2025.0   2h 41m      R       95.0     (163K)     8.0  
2            2025.0   2h 13m      R       56.0      (17K)     6.7  
3            2025.0   1h 47m  PG-13       71.0      (38K)     7.5  
4            2025.0   1h 33m  PG-13       20.0      (10K)     5.5  
...             ...      ...    ...        ...        ...     ...  
10043        1993.0   1h 37m    NaN       47.0      (18K)     5.9  
10044        2025.0   1h 33m    NaN        NaN       (81)     6.8  
10045        2023.0   1h 36m      R       51.0      (10K)     5.5  
10046        2022.0   1h 46m    NaN       63.0       (1K)     6.6  
10047        2002.0   1h 45m      R       55.0      (26K)     6.1  

[10048 rows x 10 columns]

### Thông tin cơ bản

Dữ liệu sau khi load được lưu vào biến `df`. Để có cái nhìn bao quát về cấu trúc và đặc điểm của dataset, nhóm hiển thị 5 dòng đầu và 5 dòng cuối của dữ liệu như sau 

In [5]:
# Hiển thị 5 dòng đầu
df.head()

rank                     title                               genres  \
0     1              Frankenstein       Drama, Fantasy, Horror, Sci-Fi   
1     2  One Battle After Another       Action, Crime, Drama, Thriller   
2     3           The Running Man  Action, Adventure, Sci-Fi, Thriller   
3     4        Predator: Badlands  Action, Adventure, Sci-Fi, Thriller   
4     5                  Playdate                       Action, Comedy   

                     budget  release_year run_time    mpa  metascore  \
0  $120,000,000 (estimated)        2025.0   2h 29m      R       77.0   
1  $130,000,000 (estimated)        2025.0   2h 41m      R       95.0   
2  $110,000,000 (estimated)        2025.0   2h 13m      R       56.0   
3  $105,000,000 (estimated)        2025.0   1h 47m  PG-13       71.0   
4                   $17,525        2025.0   1h 33m  PG-13       20.0   

  vote_count  rating  
0     (132K)     7.6  
1     (163K)     8.0  
2      (17K)     6.7  
3      (38K)     7.5  
4      (10K)     5.5

In [7]:
# Hiển thị 5 dòng cuối
df.tail()

rank                                title  \
10043  10046        Return of the Living Dead III   
10044  10047                            Sweetness   
10045  10048                         Dark Harvest   
10046  10049  Across the River and Into the Trees   
10047  10050                                Below   

                                  genres                   budget  \
10043                             Horror   $2,000,000 (estimated)   
10044                           Thriller                      NaN   
10045                    Fantasy, Horror                      NaN   
10046                Drama, Romance, War                      NaN   
10047  Action, Horror, Mystery, Thriller  $40,000,000 (estimated)   

       release_year run_time  mpa  metascore vote_count  rating  
10043        1993.0   1h 37m  NaN       47.0      (18K)     5.9  
10044        2025.0   1h 33m  NaN        NaN       (81)     6.8  
10045        2023.0   1h 36m    R       51.0      (10K)     5.5  
10046        2022.0   1h 46m  NaN       63.0       (1K)     6.6  
10047        2002.0   1h 45m    R       55.0      (26K)     6.1

Và số dòng, số cột của dataset

In [8]:
num_rows, num_columns = df.shape
print(f"Số dòng: {num_rows}")
print(f"Số cột: {num_columns}")

Số dòng: 10048
Số cột: 10


Dữ liệu bao gồm 10048 số lượng record và 10 đặc trưng, ý nghĩa của mỗi dòng dữ liệu phản ánh các thông tin chi tiết cho mỗi bộ phim trên nền tảng IMDb qua các đặc điểm như là xếp hạng phim, tiêu đề, thể loại, kinh phí, năm phát hành, thời lượng bộ phim, MPA, điểm metascore, số lượng bình chọn và điểm rating. 

Kích thước của bộ dữ liệu có độ lớn tương đối, có ý nghĩa đối với các phân tích xu hướng, hoặc mô hình hóa dự báo. 

### Tính toàn vẹn dữ liệu (Data integrity)

Có dòng nào trong dữ liệu bị trùng lặp (duplicated) không ? 

In [ ]:
duplicated = df.duplicated(subset = [""]).any()
print(duplicated)

### Tính chính xác dữ liệu (Data validation)

Có cột dữ liệu nào có giá trị bất hợp lý không ?

### Đánh giá mỗi cột dữ liệu

Ý nghĩa mỗi cột dữ liệu

Các cột dữ liệu nào gần với các phân tích về tiềm năng (Potential Analysis), có thể khai thác trong tương lai

Cột dữ liệu nào nên được bỏ ra ? Lí do ? 

### Kiểu dữ liệu cho mỗi đặc trưng

Thông tin cho kiểu dữ liệu hiện tại của mỗi đặc trưng ?

Có đặc trưng nào có kdl không phù hợp không ? Nếu có nên chuyển đổi thành kiểu dữ liệu nào ? 

### Phân tích trên các cột dữ liệu thuộc loại numerical

### Phân tích trên các cột dữ liệu thuộc loại categorical

### Phân tích cho missing values

### Phân tích mỗi quan hệ và tương quan 

### Nhận định sơ bộ ban đầu về dữ liệu